tests/pgautofailover_utils.py

```diff
@@ -279,93 +281,98 @@ def stop_postgres(self):
         for i in range(60):
             stop_command = [shutil.which('pg_ctl'), '-D', self.datadir,
                             '--wait', '--mode', 'fast', 'stop']
-            stop_proc = self.vnode.run(stop_command)
             try:
-                out, err = stop_proc.communicate(timeout=1)
-                break
+                with self.vnode.run(stop_command) as stop_proc:
+                    out, err = stop_proc.communicate(timeout=1)
+                    if stop_proc.returncode > 0:
+                        print("stopping postgres for '%s' failed, out: %s\n, err: %s"
+                              % (self.vnode.address, out, err))
+                        return False
+                    elif stop_proc.returncode is None:
+                        print("stopping postgres for '%s' timed out")
+                        return False
+                    return True
             except subprocess.TimeoutExpired:
+                stop_proc.release()
```

Comment:
I'm surprised that stop_proc is defined here, but maybe I'm reading the diff wrong?

Reply:
yes that's that's definitely wrong, removing it. I guess it works because `with` doesn't actually create a separate scope (just like `if`/`for`/`while` etc). And `stop_proc.release()` is probably a no-op if it's already been called (which it is because that happens when the `with` block is exited)